File: src/leetCodeProblems/MathCalculations/FindMissingNumber268.java
package leetCodeProblems.MathCalculations;

/**
 * LeetCode Link - https://leetcode.com/problems/missing-number/
 * 
 * @author anshul.agrawal
 *
 */
public class FindMissingNumber268 {
	
	public int missingNumber(int[] nums) {
        
        int arraySize = nums.length;
        int currentSum = 0;
        
        int expectedSum = (arraySize*(arraySize+1))/2;
        
        for(int i=0; i<arraySize;i++) {
            currentSum += nums[i];
        }
        
        return expectedSum - currentSum;
        
        //System.out.println("MI")
    }

	public static void main(String args[]) {

		//int[] inputArray = {3,0,1};

		int[] inputArray = {0,2};
		
		FindMissingNumber268 obj = new FindMissingNumber268();

		System.out.println(obj.missingNumber(inputArray));
		// Your code goes here
	}

}

File: src/leetCodeProblems/MathCalculations/FindMissingNumber268.go
package main

/*
- Leetcode - https://leetcode.com/problems/missing-number/
- Time - O(n)
- Space - O(1)
*/
func missingNumber(nums []int) int {
	currentSum := 0

	for _, v := range nums {
		currentSum += v
	}

	expectedSum := len(nums) * (len(nums) + 1) / 2
	return expectedSum - currentSum
}

// func main() {
// 	log.Println(missingNumber([]int{3, 0, 1}))
// 	log.Println(missingNumber([]int{9, 6, 4, 2, 3, 5, 7, 0, 1}))
// }
